tests/test_builders/test_build_linkcheck.py

```diff
@@ -145,7 +152,7 @@ def test_defaults(app):
         'info': '',
     }
 
-    def _missing_resource(filename: str, lineno: int):
+    def _missing_resource(filename: str, lineno: int) -> dict[str, str | int]:
```

Comment:
would a `TypedDict` or even a dataclass be better here?

Reply:
It would do, yep; this is not particularly robust type information.  I think I'd prefer to get some type-hints, even if non-ideal, in place and to use those to assist refactoring.

Reply:
Neither of them. It's better to have an in-place value since it doesn't appear to be used more than once.

Reply:
I'm considering re-using that helper method elsewhere - but it shouldn't be included in this changeset (1. too large a refactor, 2. I haven't explored it to know what's possible yet, 3. it would introduce conflicts elsewhere).

Reply:
If you want to reuse it, put it in an util file (and not in the test file itself if you know that it will be imported)